### DISCUSSION TITLE
Nvidia-Docker for GPU support when upload?

### DISCUSSION THREAD

**Discussion by kefhuang** [empriselab/RCareWorld#167] [Other Q&A] [answered]
Hello, I have a question regarding the evaluation platform, specifically related to GPU support. According to the website, it mentions that the platform has 12GB GPU support. However, I noticed that the [Docker template](https://github.com/empriselab/RCareWorld/blob/phy-robo-care/template/docker-template/dockerfile) provided is based on a `python:3.10-slim` image, rather than an `nvidia/cuda` base image.

In our local setup, we are utilizing an `nvidia/cudagl` base image, which requires nvidia-docker to be installed along with the `--gpus all` parameter when running the Docker container. These configurations, however, don't seem to be present in the current testing script.

Could you please confirm if using nvidia-docker is supported on your evaluation platform? If so, could you recommend the best CUDA version to use? For our local testing environment, we have found that the `nvidia/cudagl:11.4.2-devel-ubuntu20.04` image works well, supporting both PyTorch and GPU-accelerated graphics rendering.

**Comment by YoruCathy** [comment, marked as answer]
You can submit the code using `nvidia/cudagl:11.4.2-devel-ubuntu20.04` and contain the RCareWorld code in it. I think you can go ahead and do a test submission with some test code using that environment and see if we are able to run it on our end.
